### DISCUSSION TITLE
override the sort() method in the Lovata\Properties Shopaholic\Classes\Collection\PropertyValueCollection class

### DISCUSSION THREAD

**Discussion by MuninHugin** [lovata/oc-shopaholic-plugin#428] [Q&A]
Hello! Tell me if it is possible to override the sort() method in the Lovata\Properties Shopaholic\Classes\Collection\PropertyValueCollection class? This is necessary in order to change the sorting order of the properties. Now it is by value, but I need by sort_order (my custom field).
At the same time, I cannot get into the lovata/propertiesshopaholic plugin code. So far I have come to a cumbersome solution where I need to inherit a lot of classes, make my own component in my custom plugin and output a new custom component with its own methods.

**Comment by kharanenka** [comment]
Hello, You can add new custom method in PropertyValueCollection, for example customSorting(). You can find example [here](https://shopaholic.one/docs#/modules/property-value/extending/extending)

**Reply by kharanenka** [reply]
```php
$obPropertyValueList->addDynamicMethod('customSort', function () use ($obPropertyValueList) {
            
        $arResultIDList = ...list with sorted value ID's;

    return $obPropertyValueList->applySorting($arResultIDList);
});
```

**Comment by MuninHugin** [comment]
Thank You! I added it by Event::subscribe method in my Plugin
